packages/kas/src/nodes.ts

```diff
@@ -2112,15 +2171,27 @@ export class Log extends Expr {
     }
 }
 
+type TrigFunc = {
+    eval: (arg: number) => number;
+    codegen: string | ((argStr: string) => string);
+    tex: string;
+    expand?: () => Expr;
+};
+
 /* trigonometric functions */
 export class Trig extends Expr {
-    constructor(type, arg) {
+    type: string; // TODO(kevinb): Use an enum for this
```

Comment:
Do you mean an actual enum or a union? I was thinking we were generally trying to avoid enums.

Reply:
I can go with a union type.